File: apps/commerce-api/src/main/java/com/loopers/domain/commerceevent/CommerceEventService.java
package com.loopers.domain.commerceevent;

import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

import lombok.RequiredArgsConstructor;

@Component
@Transactional
@RequiredArgsConstructor
public class CommerceEventService {
    private final CommerceEventRepository commerceEventRepository;

    public void log(CommerceEventCommand.Log command) {
        CommerceEvent commerceEvent = CommerceEvent.from(command);
        commerceEventRepository.save(commerceEvent);
    }
}

File: apps/commerce-streamer/src/main/kotlin/com/loopers/domain/commerceevent/CommerceEventService.kt
package com.loopers.domain.commerceevent

import com.loopers.support.error.CoreException
import com.loopers.support.error.ErrorType
import org.springframework.stereotype.Component

@Component
class CommerceEventService(
    private val commerceEventRepository: CommerceEventRepository
) {
    fun isAlreadyProcessed(eventId: String): Boolean {
        val event = commerceEventRepository.find(eventId)
            ?: throw CoreException(ErrorType.NOT_FOUND)
        return event.processed()
    }

    fun markAsProcessed(eventId: String) {
        val event = (commerceEventRepository.find(eventId)
            ?: throw CoreException(ErrorType.NOT_FOUND))
        event.markAsProcessed();
    }
}
